test/sql/types/map/test_map_subscript.test

```diff
@@ -0,0 +1,97 @@
+# name: test/sql/types/map/test_map_subscript.test
+# description: Test cardinality function for maps
+# group: [map]
+
+# Single element on map
+query I
+select m[1] from (select MAP(LIST_VALUE(1, 2, 3, 4),LIST_VALUE(10, 9, 8, 7)) as m) as T
+----
+[10]
+
+#Element not present on MAP
+query I
+select m[0] from (select MAP(LIST_VALUE(1, 2, 3, 4,2),LIST_VALUE(10, 9, 8, 7,11)) as m) as T
+----
+[]
+
+# Null present on map
+query I
+select m[NULL] from (select MAP(LIST_VALUE(1, 2, 3, 4,2, NULL),LIST_VALUE(10, 9, 8, 7,11,42)) as m) as T
+----
+[42]
+
+# Null not present on map
+query I
+select m[NULL] from (select MAP(LIST_VALUE(1, 2, 3, 4,2),LIST_VALUE(10, 9, 8, 7,11)) as m) as T
+----
+[]
+
+#Multiple elements on map
+query I
+select m[2] from (select MAP(LIST_VALUE(1, 2, 3, 4,2),LIST_VALUE(10, 9, 8, 7,11)) as m) as T
+----
+[9, 11]
+
+# Empty maps
+query I
+select m[2] from (select MAP(LIST_VALUE(),LIST_VALUE()) as m) as T
+----
+[]
+
+query I
+select m[2] from (select MAP() as m) as T
+----
+[]
+
+# Check autocast
+query I
+select m[2::TINYINT] from (select MAP(LIST_VALUE(1, 2, 3, 4,2),LIST_VALUE(10, 9, 8, 7,11)) as m) as T
+----
+[9, 11]
+
+
+#Test Strings
+query I
+select m['Jon Lajoie'] from (select MAP(LIST_VALUE('Jon Lajoie', 'Backstreet Boys', 'Tenacious D','Jon Lajoie' ),LIST_VALUE(10,9,10,11)) as m) as T
+----
+[10, 11]
+
+query I
+select m['Spice Girls'] from (select MAP(LIST_VALUE('Jon Lajoie', 'Backstreet Boys', 'Tenacious D','Jon Lajoie' ),LIST_VALUE(10,9,10,11)) as m) as T
+----
+[]
+
+query I
+select m[NULL] from (select MAP(LIST_VALUE('Jon Lajoie', 'Backstreet Boys', 'Tenacious D','Jon Lajoie' ),LIST_VALUE(10,9,10,11)) as m) as T
+----
+[]
+
+query I
+select m[NULL] from (select MAP(LIST_VALUE('Jon Lajoie', 'Backstreet Boys', 'Tenacious D','Jon Lajoie',NULL ),LIST_VALUE(10,9,10,11,1)) as m) as T
+----
+[1]
+
+query I
+select m[0] from (select MAP(LIST_VALUE('Jon Lajoie', 'Backstreet Boys', 'Tenacious D','Jon Lajoie' ),LIST_VALUE(10,9,10,11)) as m) as T
+----
+[]
+
+query I
+select map_extract(m,1) from (select MAP(LIST_VALUE(1, 2, 3, 4),LIST_VALUE(10, 9, 8, 7)) as m) as T
+----
+[10]
+
+query I
+select map_extract(m,3) from (select MAP(LIST_VALUE(1, 2, 3, 4),LIST_VALUE(10, 9, 8, 7)) as m) as T
+----
+[8]
+
+query I
+select element_at(m,1) from (select MAP(LIST_VALUE(1, 2, 3, 4),LIST_VALUE(10, 9, 8, 7)) as m) as T
+----
+[10]
+
+query I
+select element_at(m,3) from (select MAP(LIST_VALUE(1, 2, 3, 4),LIST_VALUE(10, 9, 8, 7)) as m) as T
```

Comment:
Some more tests: 
* Map function called with more than 2 lists, or where the 2 entries are not lists, or where 1 of the 2 entries is not a list
* Map with unbalanced list cardinalities
* Map with larger-than-vector-size cardinalities
* Duplicate keys in a map
* String search with a bunch of NULL values in the map
* A map with only NULL values as key or value (or both)
* Can you struct_extract on a map? (e.g. `struct_extract(m, 'key')`)
* Searching using a non-constant value (e.g. `select m[a] from ...`)
* What about multiple rows each with a map, instead of just a single row?
* What about multiple rows each with a map, with a selection vector on top of them (e.g. because of a `WHERE` clause)?
* map_extract directly on a constant map (`SELECT map_extract(MAP(..))`, i.e. no `FROM` clause).


For the large maps, you can create lists using the `LIST` aggregate function, e.g.

```sql
SELECT LIST(i) FROM range(1000) tbl(i);
```